### PR TITLE
artifact-gen: Improve error message when mender-artifact is not found.

### PR DESCRIPTION
**Pull request by kacf**
Fixes MEN-4044.

Changelog: Title

Signed-off-by: Kristian Amlie <kristian.amlie@northern.tech>

### CODE REVIEW

File: support/modules-artifact-gen/docker-artifact-gen [hunk 1]
```diff
@@ -28,10 +28,16 @@ show_help_and_exit_error() {
...
+  fi
 }
 
-check_dependency mender-artifact
+if ! check_dependency mender-artifact; then
+  echo "Please follow the instructions here to install mender-artifact and then try again: https://docs.mender.io/downloads#mender-artifact" 1>&2
+  exit 1
+fi
 check_dependency docker
 check_dependency jq
```
[the remark this answers is not in the document]
Comment (lluiscampos) [reply support/modules-artifact-gen/docker-artifact-gen:42]: If we change this, please let's make it consistent across the three scripts.

**Review by lluiscampos — CHANGES_REQUESTED** [review]
See my comment below. I think we need to rework a bit.

File: support/modules-artifact-gen/docker-artifact-gen (hunk 1 above)
[the remark this answers is not in the document]
Comment (kacf) [reply support/modules-artifact-gen/docker-artifact-gen:42]: `set -e` is enabled, so it exits correctly on all three checks.

I think your suggested fix is indeed a bit overkill as then the message has to be repeated three times. I think the current approach is kind of Go-like, in the sense that the innermost function logs, then returns, and then the outer function may, or may not, log additional information.

**Review by lluiscampos — APPROVED** [review]
Ok. Makes sense.

**Comment by kacf** [comment]
Thanks!

**Pull request merged**